File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/DayOfWeekContextEvaluator.java
package com.github.iamrenny.ruleflow.evaluators;

import com.github.iamrenny.ruleflow.RuleFlowLanguageLexer;
import com.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import com.github.iamrenny.ruleflow.visitors.Visitor;

import java.time.LocalDateTime;

public class DayOfWeekContextEvaluator implements ContextEvaluator<RuleFlowLanguageParser.DayOfWeekContext> {

    @Override
    public Object evaluate(RuleFlowLanguageParser.DayOfWeekContext ctx, Visitor visitor) {
        Object valLeft = visitor.visit(ctx.left);
        LocalDateTime left = LocalDateTime.parse(valLeft.toString());

        if (left == null) {
            throw new IllegalArgumentException("Parameter value not supported: " + valLeft);
        }

        switch (ctx.op.getType()) {
            case RuleFlowLanguageLexer.DAY_OF_WEEK:
                return left.getDayOfWeek();
            default:
                throw new IllegalArgumentException("Operation not supported: " + ctx.op.getText());
        }
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/DayOfWeekContextEvaluator.kt
package com.rappi.analang.evaluators

import com.rappi.fraud.analang.ANALexer
import com.rappi.fraud.analang.ANAParser
import com.rappi.analang.visitors.Visitor
import java.time.LocalDateTime


class DayOfWeekContextEvaluator : ContextEvaluator<ANAParser.DayOfWeekContext> {

    override fun evaluate(ctx: ANAParser.DayOfWeekContext, visitor: Visitor): Any {
        val valLeft = visitor.visit(ctx.left)
        val left =  LocalDateTime.parse(valLeft.toString())

        if(left == null)
            throw IllegalArgumentException("parameter value not supported: $valLeft")

        return when (ctx.op.type) {
            ANALexer.DAY_OF_WEEK -> left.dayOfWeek
            else -> throw IllegalArgumentException("Operation not supported: ${ctx.op.text}")
        }

    }

}